pkg/apis/awsprovider/v1alpha1/types_test.go

```diff
@@ -0,0 +1,162 @@
+			},
+			expected: false,
+		},
+		{
+			name: "Equal should return false for different ToPort",
+			inputIngressRule: &providerv1.IngressRule{
+				Description:            "allow access from vpc",
+				Protocol:               providerv1.SecurityGroupProtocolTCP,
+				FromPort:               0,
+				ToPort:                 80,
+				CidrBlocks:             []string{"10.20.30.0/24", "10.30.40.0/24", "172.16.12.0/24"},
+				SourceSecurityGroupIDs: []string{"sg-1234567", "sg-1236745"},
+			},
+			inputIngressRuleParam: &providerv1.IngressRule{
+				Description:            "allow access from vpc",
+				Protocol:               providerv1.SecurityGroupProtocolTCP,
+				FromPort:               0,
+				ToPort:                 443,
+				CidrBlocks:             []string{"10.20.30.0/24", "10.30.40.0/24", "172.16.12.0/24"},
+				SourceSecurityGroupIDs: []string{"sg-1234567", "sg-1236745"},
+			},
+			expected: false,
+		},
+		{
+			name: "Equal should return true for same CidrBlocks",
+			inputIngressRule: &providerv1.IngressRule{
+				Description:            "allow access from vpc",
+				Protocol:               providerv1.SecurityGroupProtocolTCP,
+				FromPort:               0,
+				ToPort:                 80,
+				CidrBlocks:             []string{"10.20.30.0/24", "10.30.40.0/24", "172.16.12.0/24"},
+				SourceSecurityGroupIDs: []string{"sg-1234567", "sg-1236745"},
+			},
+			inputIngressRuleParam: &providerv1.IngressRule{
+				Description:            "allow access from vpc",
+				Protocol:               providerv1.SecurityGroupProtocolTCP,
+				FromPort:               0,
+				ToPort:                 80,
+				CidrBlocks:             []string{"172.16.12.0/24", "10.20.30.0/24", "10.30.40.0/24"},
+				SourceSecurityGroupIDs: []string{"sg-1234567", "sg-1236745"},
+			},
+			expected: true,
+		},
+		{
+			name: "Equal should return true for same SourceSecurityGroupIDs",
+			inputIngressRule: &providerv1.IngressRule{
+				Description:            "allow access from vpc",
+				Protocol:               providerv1.SecurityGroupProtocolTCP,
+				FromPort:               0,
+				ToPort:                 80,
+				CidrBlocks:             []string{"10.20.30.0/24", "10.30.40.0/24", "172.16.12.0/24"},
+				SourceSecurityGroupIDs: []string{"sg-1234567", "sg-1236745"},
+			},
+			inputIngressRuleParam: &providerv1.IngressRule{
+				Description:            "allow access from vpc",
+				Protocol:               providerv1.SecurityGroupProtocolTCP,
+				FromPort:               0,
+				ToPort:                 80,
+				CidrBlocks:             []string{"172.16.12.0/24", "10.20.30.0/24", "10.30.40.0/24"},
+				SourceSecurityGroupIDs: []string{"sg-1236745", "sg-1234567"},
+			},
+			expected: true,
+		},
+	}
+
+	for _, tc := range testcases {
+		t.Run(tc.name, func(t *testing.T) {
+			actual := tc.inputIngressRule.Equal(tc.inputIngressRuleParam)
+			if !reflect.DeepEqual(actual, tc.expected) {
+				t.Fatalf("expected equality but got\n%#v\nand expected\n%#v", actual, tc.expected)
```

Comment:
Please follow this guide for the unit test failure messages
https://github.com/golang/go/wiki/CodeReviewComments#useful-test-failures

You can maybe say `t.Fatalf("For [%s], equality check failed, Got: %t, want: %t", tc.name, actual, tc.expected)`

Reply:
Sure, will do that. Thanks.